chore/release-beta.sh

```diff
@@ -19,17 +19,45 @@ echo "NEW_BETA_NUMBER $NEW_BETA_NUMBER"
 NEXT_VERSION="$CURRENT_VERSION-beta-$NEW_BETA_NUMBER"
 echo "::endgroup::"
 
-echo "::group::Setting release version"
+BRANCH_NAME="beta-release/$NEXT_VERSION"
+
+echo "::group::Setting beta-release version"
 mvn -f spoon-pom --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_VERSION" -DprocessAllModules
 mvn --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_VERSION" -DprocessAllModules
 mvn -f spoon-javadoc --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_VERSION" -DprocessAllModules
 echo "::endgroup::"
 
-echo "::group::Staging release"
+echo "::group::Commit & Push changes"
+git checkout -b "$BRANCH_NAME"
+git commit -am "release: Releasing version $NEXT_VERSION"
+git push --set-upstream origin "$BRANCH_NAME"
+echo "::endgroup::"
+
+echo "::group::Staging beta-release"
 mvn -f spoon-pom --no-transfer-progress --batch-mode -Pjreleaser clean deploy -DaltDeploymentRepository=local::default::file:./target/staging-deploy
 mvn --no-transfer-progress --batch-mode -Pjreleaser deploy:deploy-file -Dfile="./spoon-pom/pom.xml" -DpomFile="./spoon-pom/pom.xml" -Durl="file://$(mvn help:evaluate -D"expression=project.basedir" -q -DforceStdout)/target/staging-deploy"
 echo "::endgroup::"
 
 echo "::group::Running jreleaser"
-JRELEASER_PROJECT_VERSION="$NEXT_VERSION" jreleaser-cli deploy
+JRELEASER_PROJECT_VERSION="$NEXT_VERSION" jreleaser-cli release
+echo "::endgroup::"
+
+# Set next version (patch of release version) with -SNAPSHOT suffix
+NEXT_RELEASE_VERSION=$CURRENT_VERSION_WITH_SNAPSHOT
+
+echo "::group::Updating poms to next target version"
+mvn -f spoon-pom --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_RELEASE_VERSION" -DprocessAllModules
+mvn --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_RELEASE_VERSION" -DprocessAllModules
+mvn -f spoon-javadoc --no-transfer-progress --batch-mode versions:set -DnewVersion="$NEXT_RELEASE_VERSION" -DprocessAllModules
+echo "::endgroup::"
```

Comment:
As far as I can see you do not actually update anything as `NEXT_RELEASE_VERSION` is already identical to whatever was there originally?

Reply:
`NEXT_VERSION` is the beta version for which the commit needs to be pushed. So I push the commit, after rewritting POMs, in R30-R42.

Next is `NEXT_RELEASE_VERSION` which is the SNAPSHOT. It will be the same version before the beta version as we don't have SNAPSHOT versions for beta as well. The name of this variable is confusing, a better name would be `SNAPSHOT_VERSION_TO_RESET_TO`.

P.S. Sorry I did not notice your reviews earlier.